mesonbuild/environment.py

```diff
@@ -1535,7 +1537,10 @@ def _detect_objc_or_objcpp_compiler(self, for_machine: MachineInfo, objc: bool)
                     exe_wrap, defines, linker=linker)
             if 'clang' in out:
                 linker = None
-                comp = ClangObjCCompiler if objc else ClangObjCPPCompiler
+                if 'Apple' in out:
+                    comp = AppleClangObjCCompiler if lang == 'c' else AppleClangObjCPPCompiler
```

Comment:
There's no 'lang' defined here. That's causing the CI failure.

Reply:
oops, you're right of course, the objc variant uses a bool. Which I should know because I wrote...